python/ray/data/datasource/image_folder_datasource.py

```diff
@@ -133,12 +130,10 @@ def _read_file(self, f: "pyarrow.NativeFile", path: str, **reader_args):
             # Try to convert image ndarray to TensorArrays.
             image = TensorArray([np.array(image)])
```

Comment:
Our options are:
1. Add the column as a list of ndarrays, i.e. `[np.array(image)]`, and let the Pandas block do the cast.
2. Explicitly do the cast in the datasource.

I was thinking that (2) would be better since we could give a more informative error and it would reduce the number of image array copies, but the latter point is moot since this is a single-ndarray tensor column (the downstream cast in the Pandas block will be zero-copy).

Do you think the more informative/specific error is worth it?

Reply:
ok sounds good to have a more informative error!

Reply:
regarding 2) isn't the exception message here same as `_cast_ndarray_columns_to_tensor_extension` ?

Reply:
@jiaodong Only difference is saying "failed to cast the image ndarrays", although I suppose the benefit is minimal since the failed column cast would be for the `"image"` column... Maybe this one can be consolidated into the generic block cast.

Reply:
Decided to stick with an explicit cast since we're explicitly not allowing heterogeneous-shaped images at the moment, so always casting to our tensor extension type is desirable.

Reply:
Decided to go with the Pandas block cast so we can technically support heterogeneous-shaped images in this datasource.